packages/lesswrong/lib/utils/markdownItPlugins.ts

```diff
@@ -47,10 +49,74 @@ export function getMarkdownItArbital() {
 }
 
 let _mdiNoPlugins: markdownIt|null = null;
-export function getMarkdownItNoPlugins() {
+export function getMarkdownItNoPlugins(): markdownIt {
   if (!_mdiNoPlugins) {
     const mdi = markdownIt({linkify: true})
     _mdiNoPlugins = mdi;
   }
   return _mdiNoPlugins;
 }
+
+
+function applyMarkdownFootnoteRules(mdi: markdownIt) {
+  // These rules were generated by starting from the rules in the source of
+  // markdown-it-footnote (https://github.com/markdown-it/markdown-it-footnote/blob/master/index.mjs)
+  // and modifying until they matched the classnames and format of our CkEditor
+  // footnote plugin.
+  mdi.renderer.rules.footnote_ref = (tokens, idx, options, env, self) => {
+    const id = self.rules.footnote_anchor_name!(tokens, idx, options, env, self)
+    const caption = self.rules.footnote_caption!(tokens, idx, options, env, self)
+    let refid = id
+  
+    if (tokens[idx].meta.subId > 0) refid += `:${tokens[idx].meta.subId}`
+  
+    //return `<sup class="footnote-ref"><a href="#fn${id}" id="fnref${refid}">${caption}</a></sup>`
```

Comment:
nit: cleanup

Reply:
These were left behind on purpose (to show what the markdown-it-footnote default was for comparison). Added an explanatory comment to that effect.